ui/src/datatypes.ts

```diff
@@ -11,25 +11,25 @@ export const all: DataTypeOption[] = [
-  { name: 'long', uri: expand('xsd:long'), params: defaultParams },
-  { name: 'int', uri: expand('xsd:int'), params: defaultParams },
-  { name: 'short', uri: expand('xsd:short'), params: defaultParams },
-  { name: 'byte', uri: expand('xsd:byte'), params: defaultParams },
-  { name: 'nonNegativeInteger', uri: expand('xsd:nonNegativeInteger'), params: defaultParams },
-  { name: 'positiveInteger', uri: expand('xsd:positiveInteger'), params: defaultParams },
-  { name: 'unsignedLong', uri: expand('xsd:unsignedLong'), params: defaultParams },
-  { name: 'unsignedInt', uri: expand('xsd:unsignedInt'), params: defaultParams },
-  { name: 'unsignedShort', uri: expand('xsd:unsignedShort'), params: defaultParams },
-  { name: 'unsignedByte', uri: expand('xsd:unsignedByte'), params: defaultParams },
-  { name: 'nonPositiveInteger', uri: expand('xsd:nonPositiveInteger'), params: defaultParams },
-  { name: 'negativeInteger', uri: expand('xsd:negativeInteger'), params: defaultParams },
-  { name: 'number', uri: expand('xsd:double'), params: defaultParams },
+  { name: 'decimal', uri: expand('xsd:decimal'), params: [...defaultParams, 'format'] },
+  { name: 'integer', uri: expand('xsd:integer'), params: [...defaultParams, 'format'] },
+  { name: 'long', uri: expand('xsd:long'), params: [...defaultParams, 'format'] },
+  { name: 'int', uri: expand('xsd:int'), params: [...defaultParams, 'format'] },
+  { name: 'short', uri: expand('xsd:short'), params: [...defaultParams, 'format'] },
+  { name: 'byte', uri: expand('xsd:byte'), params: [...defaultParams, 'format'] },
+  { name: 'nonNegativeInteger', uri: expand('xsd:nonNegativeInteger'), params: [...defaultParams, 'format'] },
+  { name: 'positiveInteger', uri: expand('xsd:positiveInteger'), params: [...defaultParams, 'format'] },
+  { name: 'unsignedLong', uri: expand('xsd:unsignedLong'), params: [...defaultParams, 'format'] },
+  { name: 'unsignedInt', uri: expand('xsd:unsignedInt'), params: [...defaultParams, 'format'] },
+  { name: 'unsignedShort', uri: expand('xsd:unsignedShort'), params: [...defaultParams, 'format'] },
+  { name: 'unsignedByte', uri: expand('xsd:unsignedByte'), params: [...defaultParams, 'format'] },
+  { name: 'nonPositiveInteger', uri: expand('xsd:nonPositiveInteger'), params: [...defaultParams, 'format'] },
+  { name: 'negativeInteger', uri: expand('xsd:negativeInteger'), params: [...defaultParams, 'format'] },
+  { name: 'number', uri: expand('xsd:double'), params: [...defaultParams, 'format'] },
   { name: 'duration', uri: expand('xsd:duration'), params: [...defaultParams, 'format'] },
   { name: 'dayTimeDuration', uri: expand('xsd:dayTimeDuration'), params: [...defaultParams, 'format'] },
   { name: 'yearMonthDuration', uri: expand('xsd:yearhMonthDuration'), params: [...defaultParams, 'format'] },
-  { name: 'float', uri: expand('xsd:float'), params: defaultParams },
+  { name: 'float', uri: expand('xsd:float'), params: [...defaultParams, 'format'] },
```

Comment:
Looks to me like a `const numericParams = [...defaultParams, 'format']` :neckbeard:

Reply:
Hm, on second look there is a lot of array spreading here. Quite unreadable actually

Reply:
I refactored the whole thing... but not sure if it's better or worst 😆